src/shared/lib/date/formatDate.ts

```diff
@@ -0,0 +1,22 @@
+import { DateTime } from 'luxon';
+
+type Format = keyof typeof formatMap;
+
+const formatMap = {
+  time: DateTime.TIME_WITH_SECONDS,
+  dateWithWeekDay: { ...DateTime.DATE_MED_WITH_WEEKDAY },
+};
+
+export const formatDate = (date: Date, format: Format) => {
+  const parsedDate = DateTime.fromJSDate(date);
+
+  if (format === 'dateWithWeekDay') {
+    const { days: diffDays } = parsedDate.diffNow('days');
+
+    if (diffDays >= -2) {
+      return parsedDate.toRelativeCalendar()?.toUpperCase() || '';
```

Comment:
Acho que esse chaining faz mais sentido, né? Afinal se parsedDate não existir daria que `toUpperCase()` não é uma function de undefined
E sem esse primeiro `?.` daria erro no `toRelativeCalendar()`, não?
```suggestion
      return parsedDate?.toRelativeCalendar()?.toUpperCase() || '';
```